joomla/joomla-cms
# [4.0] Module 'Articles - Newsflash' - exclude current article

This PR adds in the **Module Articles - Newsflash** (mod_articles_news) an option exclude the current article to filter the current article from displaying in the newsflash module..
It's PR #23575 "Exclude current article"  re-done for Joomla 4. 

### Summary of Changes
The **Module Articles - Newsflash** can display the same article as the article currently displayed on the page.

- This  PR adds an option in the Latest News Module to exclude the current article.
- The option is OFF by default (so the current behavior before installing the patch is default)
- It also checks for com_content being active to prevent that other components that use an "id" might interfere with the Latest News Module output.

### Testing Instructions
- Create a **Module Articles - Newsflash**, choose Category :Blog", assign the module to position "sidebar-right" and assign the module to display "On all pages". 
- Go to a page that displays an article, for example index.php/blog/3-welcome-to-your-blog
- Refresh the page a couple of times to see that the Newsflash Module might display the same article text.

![mod_articles_newsflash-before-frontend](https://user-images.githubusercontent.com/1217850/56475143-a2b8ee00-6484-11e9-9ca2-4e6a70100ba3.png)

### Expected result
Before installing this PR, the following options are available:

![mod_articles_newsflash-before-backend](https://user-images.githubusercontent.com/1217850/56475140-a2b8ee00-6484-11e9-9d36-fda63e9a2645.png)

After installing this PR, there's a new option available: "Exclude current Article" that is set to "No" by default.

![mod_articles_newsflash-after-no](https://user-images.githubusercontent.com/1217850/56475142-a2b8ee00-6484-11e9-8ed6-daf078aaa75a.png)

After configuring the "Exclude current Article" to "Yes", the module should not display the same article anymore.

![mod_articles_newsflash-after-yes](https://user-images.githubusercontent.com/1217850/56475141-a2b8ee00-6484-11e9-843d-da4e1227e8a4.png)
